src/v/raft/consensus.cc

```diff
@@ -722,30 +720,38 @@ consensus::linearizable_barrier(model::timeout_clock::time_point deadline) {
         update_node_append_timestamp(target);
         vlog(
           _ctxlog.trace, "Sending empty append entries request to {}", target);
-        auto f = _client_protocol
-                   .append_entries(
-                     target.id(),
-                     std::move(req),
-                     rpc::client_opts(_replicate_append_timeout))
-                   .then([this, id = target.id(), seq, dirty_offset](
-                           result<append_entries_reply> reply) {
-                       process_append_entries_reply(
-                         id, reply, seq, dirty_offset);
-                   });
-
-        send_futures.push_back(std::move(f));
+
+        auto holder = _bg.hold();
+        ssx::background
+          = ssx::spawn_with_gate_then(
+              _bg,
+              [this,
+               target,
+               req = std::move(req),
+               seq,
+               dirty_offset]() mutable {
+                  return _client_protocol
+                    .append_entries(
+                      target.id(),
+                      std::move(req),
+                      rpc::client_opts(_replicate_append_timeout))
+                    .then([this, id = target.id(), seq, dirty_offset](
+                            result<append_entries_reply> reply) {
+                        process_append_entries_reply(
+                          id, reply, seq, dirty_offset);
+                    });
+              })
+              .handle_exception([holder = std::move(holder),
+                                 this](const std::exception_ptr& e) {
+                  vlog(_ctxlog.warn, "append entries dispatch failed: {}", e);
+              });
```

Comment:
can this handle exception be moved inside the `spawn_with_gate`. This way we will not need another gate holder as the exception will be handed while holding the gate already

Reply:
we could also add some details such as target id into the warning message

Reply:
good idea, done. I put a holder here so it throws if the gate is closed by this point but I think equivalent behavior can be achieved with a gate.check() right before this loop and skipping a holder creation here.